src/solvers/dgsem_tree/dg_2d_parabolic.jl

```diff
@@ -0,0 +1,652 @@
+# This file collects all methods that have been updated to work with parabolic systems of equations
+#
+# assumptions: parabolic terms are of the form div(f(u, grad(u))) and
+# will be discretized first order form as follows:
+#               1. compute grad(u)
+#               2. compute f(u, grad(u))
+#               3. compute div(u)
+# boundary conditions will be applied to both grad(u) and div(u).
+function rhs_parabolic!(du, u, t, mesh::TreeMesh{2}, equations_parabolic::AbstractEquationsParabolic,
+                        initial_condition, boundary_conditions, source_terms,
+                        dg::DG, dg_parabolic, cache, cache_parabolic)
+
+  # Reset du
+  @trixi_timeit timer() "reset ∂u/∂t" reset_du!(du, dg, cache)
+
+
+  @unpack u_transformed, u_grad, viscous_flux = cache_parabolic
+  @trixi_timeit timer() "transform variables" transform_variables!(u_transformed, u,
+                                                                   mesh, equations_parabolic,
+                                                                   dg, dg_parabolic,
+                                                                   cache, cache_parabolic)
+
+  @trixi_timeit timer() "calculate gradient" calc_gradient!(u_grad, u_transformed, t, mesh,
```

Comment:
How would you call it in 1D for a scalar equation?

Reply:
I consider 1D an edge case so I could live with this 😎. But OTOH, we can leave it as it is (it can be interpreted as "calculate the gradient for all variables").